### PR TITLE
Add operation to list SAML provider configs.

### PR DESCRIPTION
**Pull request by micahstairs**
Adds a list operation for SAML provider configs. I've included both unit tests and integration tests.

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java [hunk 1]
```diff
@@ -1335,6 +1336,93 @@ protected SamlProviderConfig execute() throws FirebaseAuthException {
...
+   *
+   * @param pageToken A non-empty page token string, or null to retrieve the first page of provider
+   *     configs.
+   * @return A {@link ListProviderConfigsPage} instance.
+   * @throws IllegalArgumentException If the specified page token is empty
+   * @throws FirebaseAuthException If an error occurs while retrieving provider config data.
+   */
+  public ListProviderConfigsPage<SamlProviderConfig> listSamlProviderConfigs(
+        @Nullable String pageToken) throws FirebaseAuthException {
+    int maxResults = FirebaseUserManager.MAX_LIST_PROVIDER_CONFIGS_RESULTS;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java:1351]:
```
return listSamlProviderConfigs(pageToken, FirebaseUserManager.MAX_LIST_PROVIDER_CONFIGS_RESULTS);
```

File: src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+public final class ListSamlProviderConfigsResponse
+    implements ListProviderConfigsResponse<SamlProviderConfig> {
+
+  @Key("inboundSamlConfigs")
+  private List<SamlProviderConfig> providerConfigs;
+
+  @Key("nextPageToken")
+  private String pageToken;
+
+  public ListSamlProviderConfigsResponse() { }
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java:37]: Unnecessary?

File: src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  @Override
+  public boolean hasProviderConfigs() {
+    return providerConfigs != null && !providerConfigs.isEmpty();
+  }
+
+  @Override
+  public String getPageToken() {
+    return pageToken == null ? "" : pageToken;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java:51]: Strings.nullToEmpty()

File: src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java (hunk 1 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java]: Done.

File: src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java (hunk 2 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java]: Removed.

File: src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java (hunk 3 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/internal/ListSamlProviderConfigsResponse.java]: Nice! That's cleaner.

File: src/main/java/com/google/firebase/auth/internal/ListOidcProviderConfigsResponse.java [hunk 4]
```diff
@@ -36,14 +34,6 @@
...
 
-  @VisibleForTesting
-  public ListOidcProviderConfigsResponse(
-      List<OidcProviderConfig> providerConfigs,
-      String pageToken) {
-    this.providerConfigs = providerConfigs;
-    this.pageToken = pageToken;
-  }
-
   public ListOidcProviderConfigsResponse() { }
```
[the remark this answers is not in the document]
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/internal/ListOidcProviderConfigsResponse.java]: Good point! Done.

**Pull request merged**